pkgs/development/libraries/libdigidocpp/default.nix

```diff
@@ -10,13 +10,18 @@ stdenv.mkDerivation rec {
      sha256 = "sha256-U5i5IAyJF4359q6M6mQemEuG7+inPYIXqLy8GHv4dkg=";
   };
 
-  nativeBuildInputs = [ cmake pkg-config xxd ];
+  nativeBuildInputs = [ cmake makeWrapper pkg-config xxd ];
 
   buildInputs = [
     minizip pcsclite opensc openssl xercesc
     xml-security-c xsd zlib xalanc
   ];
 
+  postInstall = ''
+    wrapProgram $out/bin/digidoc-tool \
+      --prefix LD_LIBRARY_PATH : ${opensc}/lib/pkcs11/
```

Comment:
Prefixing `LD_LIBRARY_PATH` is messy, as it propagates processes. Check if you can set `PKCS11_ENGINE=` to an absolute path (like in `pkgs/tools/security/rng-tools/default.nix`), or if it's possible to update the path that's loaded at runtime to an absolute one (like in `pkgs/tools/security/chrome-token-signing/default.nix`).

If one of these is possible, the same approach should be applied to `pkgs/tools/security/qdigidoc/default.nix`, too.

Reply:
In any case, this should at least include a comment with a link to the upstream PR, so people can track the upstreaming status.

In the case of this application here, all that's needed should be a

```
+  cmakeFlags = [
+    "-DPKCS11_MODULE=${opensc}/lib/opensc-pkcs11.so"
+  ];
```

This currently doesn't work on Linux, as the upstream `FindPKCS11.cmake` unconditionally sets the value - I sent a PR in https://github.com/open-eid/cmake/pull/35 to fix this, which seems a bit cleaner than just setting a random environment variable during the build.

Reply:
I'd say `PKCS11_MODULE_PATH` is as random as `PKCS11_MODULE`...

Both PRs work as expected and I don't mind either way, so I'll leave that to upstream.

I'll add a link so nixpkgs hackers will see it, although I dare say one ought to look through upstream changes anyway
when updating a package (just like basic functionality should be tested....).